Userland/Applications/PixelPaint/Tools/GradientTool.cpp

```diff
@@ -346,32 +425,71 @@ void GradientTool::reset()
     m_gradient_start = {};
     m_gradient_center = {};
     m_gradient_end = {};
+    m_gradient_horizontal_a = {};
+    m_gradient_horizontal_b = {};
     m_gradient_half_length = 0;
     m_physical_diagonal_layer_length = 0;
     m_hover_over_drag_handle = false;
     m_hover_over_start_handle = false;
     m_hover_over_end_handle = false;
+    m_hover_over_horizontal_a_handle = false;
+    m_hover_over_horizontal_b_handle = false;
 
     if (m_editor) {
         m_editor->update();
         m_editor->update_tool_cursor();
     }
 }
 
-void GradientTool::update_gradient_end_and_derive_start(Gfx::IntPoint const new_end_point)
+void GradientTool::update_gradient_with_initial_values(Gfx::IntPoint const new_end_point)
 {
     VERIFY(m_gradient_center.has_value());
     m_gradient_end = new_end_point;
-    m_gradient_start = m_gradient_center.value() - (m_gradient_end.value() - m_gradient_center.value());
+    auto deltaCenter = m_gradient_end.value() - m_gradient_center.value();
+    m_gradient_start = m_gradient_center.value() - deltaCenter;
+
+    if (m_mode == GradientMode::Radial) {
+        Gfx::IntPoint perpendicularDeltaCenter = { -deltaCenter.y(), deltaCenter.x() };
+        m_gradient_horizontal_a = m_gradient_center.value() + perpendicularDeltaCenter;
+        m_gradient_horizontal_b = m_gradient_center.value() - perpendicularDeltaCenter;
+    }
 }
 
-void GradientTool::translate_gradient_start_end(Gfx::IntPoint const delta, bool update_start_counterwise)
+void GradientTool::translate_gradient_points(Gfx::IntPoint const delta, TranslationType translation_type)
 {
     m_gradient_end.value().translate_by(delta);
-    if (update_start_counterwise)
+    auto distance_delta = m_gradient_center.value().distance_from(m_gradient_end.value()) - m_gradient_center.value().distance_from(m_gradient_start.value());
```

Comment:
itʼs just `distance`. to be more specific you might even settle for more descriptive name, e.g. `gradient_length`?

Reply:
i changed it to `translation_distance_to_center`  as this is the change of distance to the center which is taken into account when calculating the transversal points.